mkosi/kmod.py

```diff
@@ -180,6 +180,11 @@ def gen_required_kernel_modules(
             mods = set(modulesd.rglob("*.ko*"))
         firmware = set()
 
+    links = {
+        p.resolve().relative_to(root)
+        for p in filter(lambda fw: fw.is_symlink() and fw.exists(), map(lambda fw: root / fw, firmware))
```

Comment:
Please use list comprehensions or a regular for loop, we don't use `filter()` or `map()` anywhere else, let's keep it that way for consistency.